src/components/forms/FileInput/FileInput.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from 'react'
+import classnames from 'classnames'
+
+import { FilePreview } from './FilePreview'
+import { makeSafeForID } from './utils'
+
+interface FileInputProps {
+  id: string
+  name: string
+  disabled?: boolean
+  multiple?: boolean
+  accept?: string
+}
+
+export const FileInput = (
+  props: FileInputProps & JSX.IntrinsicElements['input']
+): React.ReactElement => {
+  const {
+    name,
+    id,
+    disabled,
+    multiple,
+    className,
+    accept,
+    ...inputProps
+  } = props
+  const [isDragging, setIsDragging] = useState(false)
+  const [showError, setShowError] = useState(false)
+  const [files, setFiles] = useState<FileList | null>(null)
+
+  const fileInputClasses = classnames(
+    'usa-file-input',
+    {
+      'usa-file-input--disabled': disabled,
+    },
+    className
+  )
+
+  const targetClasses = classnames('usa-file-input__target', {
+    'usa-file-input--drag': isDragging,
+    'has-invalid-file': showError,
+  })
+
+  const hideDragText =
+    /rv:11.0/i.test(navigator.userAgent) ||
+    /Edge\/\d./i.test(navigator.userAgent)
+
+  const dragText = multiple ? 'Drag files here or ' : 'Drag file here or '
+
+  const filePreviews = []
+  if (files) {
+    for (let i = 0; i < files?.length; i++) {
+      const imageId = makeSafeForID(files[i].name)
+      const key = `filePreview_${imageId}`
+      filePreviews.push(
+        <FilePreview key={key} imageId={imageId} file={files[i]} />
```

Comment:
I wonder if we should surface an optional prop for FilePreview  (defaulting to this component) so people could build their own?

Reply:
this feels like an enhancement that goes beyond the USWDS implementation, mind adding as a new feature request?

Reply:
Good point https://github.com/trussworks/react-uswds/issues/957